ENH: update auth process to stay compliant

deniederhut opened the pull request:
We had been using a simple account name and password to log in
directly to Gmail's smtp server, but Google no longer supports this
as of today https://support.google.com/accounts/answer/6010255?hl=en

The mail script has been rewritten to use an application-based,
assumed role OAuth workflow.

deniederhut committed:
ENH: update auth process to stay compliant

We had been using a simple account name and password to log in
directly to Gmail's smtp server, but Google no longer supports this
as of today https://support.google.com/accounts/answer/6010255?hl=en

The mail script has been rewritten to use an application-based,
assumed role OAuth workflow.

scoobies commented:
Hi!, I'm @scoobies, a friendly bot.

Type ```@scoobies help``` to discover how I can help you.

deniederhut committed:
ENH: new email templates

I made these last year, but I guess they never made their
way into dev.

deniederhut committed:
ENH: update deadlines

deniederhut committed:
ENH: email template updates

mepa commented on publisher/mail/_mailer.py:
```diff
@@ -1,12 +1,14 @@
+from google.oauth2.credentials import Credentials
+from google_auth_oauthlib.flow import InstalledAppFlow
+from googleapiclient.discovery import build, Resource
+from googleapiclient.errors import HttpError
```
These google packages seem likely to be backward compatible over time. However, if convenient, adding a requirements.txt of packages with version numbers may help a future somebody set up a working environment for sending emails.

deniederhut replied:
I couldn't recover the versions I was using -- different year, different laptop, etc. But I added a reqfile with at least the top level dependencies.

mepa commented on publisher/mail/mail_review_final.py:
```diff
@@ -0,0 +1,76 @@
+    # we make this table by hand in google sheets when the committee divvies
+    # up reviewers among papers -- the format we have been using is something like
+    # | paper title | PR url | reviewer one | reviewer two | ... | reviewer n |
+    # where each reviewer field is like "firstname lastname <email@provider>"
```
Great context - thanks for adding these comments

mepa reviewed (approved):
LGTM. Thanks for updating this and sorry for not taking a look earlier.

deniederhut committed:
ENH: add reqfile

Also fix bug with escape function that was moved to
a different module in python 3.8
